### PR TITLE
Extend col types

### PR DESCRIPTION
**Pull request by dtassone**

### CODE REVIEW

File: packages/storybook/src/documentation/pages/columns.stories.mdx [hunk 1]
```diff
@@ -66,7 +67,55 @@ We also added locale formatting for date, dateTime, and number Columns.
...
+### Extending column types
+
+XGrid lets you extend the native column types with your own columns types.
+To do this, you need to define your new column type first and add it to `options.columnTypes` as below.
+
+**Note** `ColTypeDef` lets you extend native columnTypes by defining the `extendType` property.
+If you do not define this property, then your new column type will extend the default column type of `string`.
+
+```tsx
+//We define 2 new column types.
```
Comment (oliviertassinari) [inline packages/storybook/src/documentation/pages/columns.stories.mdx:79]: Space after `//` (and in the other places).

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 2]
```diff
@@ -194,6 +195,11 @@ export interface GridOptions {
    * Set of icons used in the grid
    */
   icons: IconsOptions;
+
+  /**
+   * Extend native column types with your new column types
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:200]: dot at the end :)

File: packages/storybook/src/documentation/pages/demos/columns/extendColTypes.demo.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+const highlightCellColType: ColTypeDef = { cellClassName: 'highlight' };
+const usdPriceColType: ColTypeDef = {
+  extendType: 'number',
+  width: 150,
+  valueFormatter: ({ value }) => `${Number(value).toFixed(2)} USD`,
+};
+
+export default function ExtendColTypesDemo() {
+  const columns: Columns = [
```
Comment (oliviertassinari) [inline packages/storybook/src/documentation/pages/demos/columns/extendColTypes.demo.tsx:14]: What do you think about moving these variables in the outer scope? It I wonder about the need to create them at each render. In the docs, there are action that can recreate them, so in people's code, maybe that could help performance a bit?

Comment (dtassone) [reply packages/storybook/src/documentation/pages/demos/columns/extendColTypes.demo.tsx:14]: Up to you. In the real world, they will be defined inside the component scope.
Performance-wise it's negligible.

Comment (oliviertassinari) [reply packages/storybook/src/documentation/pages/demos/columns/extendColTypes.demo.tsx:14]: OK, no preferences, I was assuming that the configuration will 80% of the time be static. But I don't have the whole context on how it's used. In this case, you might want to revert some of my previous changes where I moved the variables outside of the render scope'

Comment (dtassone) [reply packages/storybook/src/documentation/pages/demos/columns/extendColTypes.demo.tsx:14]: It seems ok to me. I checked the demos

**Comment by dtassone** [comment]
Merging! I will have to fix the versioning issue in master now

**Pull request merged**